src/ctia/http/routes/common.clj

```diff
@@ -50,6 +53,65 @@
    (s/optional-key :search_after) (describe [s/Str] "Pagination stateless cursor")
    (s/optional-key :limit) (describe Long "Pagination Limit")})
 
+(def default-ignored-search-fields
+  #{:language
+    :revision
+    :schema_version
+    :source
+    :source_uri
+    :timestamp
+    :tlp
+    :type})
+
+(def default-whitelisted-search-fields
+  #{:id
+    :title
+    :short_description
+    :description
+    :external_ids})
+
+(s/defn searchable-fields :- (s/protocol s/Schema)
+  [{:keys [fields ignore]}]
```

Comment:
why not simply provide the list of searchable fields?
``` clojure
(routes.common/searchable-fields
                               (dissoc (set actor-fields)
                                            :valid_time.start_time
                                            :valid_time.end_time))
```

Reply:
> why not simply provide the list of searchable fields?

If you remember, I didn't want us to repeat the list of allowed and blacklisted fields for every single entity, 

So it doesn't look like this (in this case for Actor):

    (routes.common/searchable-fields
     {:allowed-fields #{:actor_type
                        :confidence
                        :description
                        :external_ids
                        :id
                        :intended_effect
                        :motivation
                        :short_description
                        :sophistication
                        :title}
      :ignored-fields [:language
                       :revision
                       :schema_version
                       :source
                       :source_uri
                       :timestamp
                       :tlp
                       :type
                       :valid_time.end_time
                       :valid_time.start_time]})

Since we can have a global default whitelist and a global ignore list, we can just use them, and for every entity we can specifically tell which fields we want to whitelist or blacklist à la carte.

But I wanted to make it clear which fields (in addition to default ignore list) need to be ignored. So the function takes a list of fields, and [optionally] fields to ignore.

I guess I need to add some proper docstrings.

Reply:
I properly understood the process. I am ok with this implementation, but you pass a first collection of fields to add and then remove some of these fields, so IMHO simply pass the search-fields without those that should be ignored. That's the responsibility of the caller to only pass relevant search fields (and you automatically ensure the presence / absence of common fields if necessary).

there are:
- default field to include
- default field to exclude
- specific fields to add

So roughly I would expect
``` clojure
(difference (union default-whitelisted-search-fields search-fields)
                   default-ignored-search-fields)
```

Reply:
> simply pass the search-fields without those that should be ignored.

I thought about it. In that case how do we precisely determine which fields should be ignored? If we rely on `default-ignored-search-fields`, then there's no way to explicitly tell not to ignore any specific field. Then we'd say: "what if we remove default-ignored-fields", in that case, for every entity we would have to explicitly list them, like so:

```
(routes.common/searchable-fields (dissoc weakness-fields :language
                                                         :revision
                                                         :schema_version
                                                         :source
                                                         :source_uri
                                                         :timestamp
                                                         :tlp
                                                         :type))
```

Or always ignore the fields in `default-ignored-fields` without a way of overriding that.